src/BinaryParsers/PEBinary/ProgramDatabase/CompilerCommandLine.cs

```diff
@@ -100,6 +106,17 @@ public CompilerCommandLine(string commandLine)
                                 this.WarningLevel = wChar - '0';
                             }
                         }
+                        else if (argument.EndsWith("O1") || argument.EndsWith("O2") || argument.EndsWith("Og") || argument.EndsWith("Os") || argument.EndsWith("Ot") || argument.EndsWith("Ox"))
+                        {
+                            // https://docs.microsoft.com/cpp/build/reference/o-options-optimize-code?view=msvc-170
+                            // /O1 /O2 /Og /Os /Ot /Ox are all indicative of optimizations being enabled
+                            this.OptimizationsEnabled = true;
+                        }
+                        else if (argument.EndsWith("Od"))
```

Comment:
>argument [](http://example.com/codeflow?start=33&length=8)

So what about command-line argument order-sensitivity? Will the compiler error out if I try to optimize with /O1 and also provide /Od? (so that we have a guarantee these two conditions are exclusive between successful compilations). Or does /Od override or not depending on position? Etc. 

The same question exists for the other functionality in this method, I have suspicious we should review the warning level handling.

But since you're code is new and this looks pretty easy to test, I'd like to dig into that for the optimizations settings.

Reply:
It warns if you combine them (VS2022 toolchain):
`1>cl : command line warning D9025: overriding '/O2' with '/Od'`
`1>cl : command line warning D9025: overriding '/Od' with '/O2'`

The second option seems to win.  As a result I _think_ this is correct because whichever comes second will set `this.OptimizationsEnabled` and that is what the compiler will be doing.

Reply:
Yes, makes sense and consistent with some of your other testing. As long as our logic for exclusive conditions consistently sets all relevant state in each case, we can assume we'll exit parsing in the same condition as the compiler, in terms of assumed configuration.